lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3983,6 +3983,15 @@ ParseResult FIRCircuitParser::parseCircuit(
   if (foundUnappliedAnnotations)
     return failure();
 
+  // If the module has an entry point that is not an external module, set the
+  // visibility of all non-main modules to private.
+  auto main = circuit.getMainModule();
+  if (isa<FModuleOp>(main)) {
+    for (auto mod : circuit.getOps<FModuleLike>()) {
```

Comment:
Although I don't know that it matters, this sets external modules to private, which seems weird for a declaration.
followup would be to mark external modules as isDeclaration.

Reply:
Thanks for pointing this out - I still think it makes sense for them to be private declarations, similarly to how hidden visibility works with ELF. I'll add the declaration flag once we progress with the linker.